exercises/twelve-days/example.js

```diff
@@ -0,0 +1,55 @@
+var TwelveDays = function() {
+  var instance = this;
+
+  (function() {
+    instance.verseList = [
+      'On the first day of Christmas my true love gave to me, a Partridge in a Pear Tree.',
+      'On the second day of Christmas my true love gave to me, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the third day of Christmas my true love gave to me, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the fourth day of Christmas my true love gave to me, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the fifth day of Christmas my true love gave to me, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the sixth day of Christmas my true love gave to me, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the seventh day of Christmas my true love gave to me, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the eighth day of Christmas my true love gave to me, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the ninth day of Christmas my true love gave to me, nine Ladies Dancing, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the tenth day of Christmas my true love gave to me, ten Lords-a-Leaping, nine Ladies Dancing, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the eleventh day of Christmas my true love gave to me, eleven Pipers Piping, ten Lords-a-Leaping, nine Ladies Dancing, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the twelfth day of Christmas my true love gave to me, twelve Drummers Drumming, eleven Pipers Piping, ten Lords-a-Leaping, nine Ladies Dancing, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+    ];
+  })();
+
+  instance.startFromZero = function(oneIndexArray) {
```

Comment:
Is there a reason you decided to declare these methods as properties of the instance rather than on the TwelveDays prototype? EG:

```javascript
function TwelveDays(){};

TwelveDays.prototype.verseList = [...]
TwelveDays.prototype.startFromZero = function(){...}
```

Either will work in this case, but they're not equivalent.  Just wondering what your thinking was?

Reply:
I didn't see this pattern to create constructor functions. @AdhirRamjiawan, do you have any reference about it? It'd be interesting to read a bit about it.

I don't know about its pros and cons, but I would feel more confortable implementing it as @matthewmorgan suggests

Reply:
@matthewmorgan , nope, in this case no particular reason why I've created them as instance methods. I've just been used to creating functions in this way. I will change it to use the prototype.

@rchavarria I'll try to look for some sites to ref this however it's just been something I've been doing for a while now. When working with some front end frameworks like Angular/Jquery I occasionally found myself needing to preserve "this" in the scope it's in especially when nesting functions. I just use it as a way to get functionality into a JS object without having any unintended side effects with using the prototype chain.